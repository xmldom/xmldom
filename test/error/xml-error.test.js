'use strict';

const { getTestParser } = require('../get-test-parser');

describe('html vs xml:', () => {
	it.each(['text/html', 'text/xml'])('unclosed document in %s', (mimeType) => {
		const { errors, parser } = getTestParser();

		const actual = parser.parseFromString('<img>', mimeType).toString();

		expect({ actual, ...errors }).toMatchSnapshot();
	});

	it.each([
		['<test><!--', '<test/>'],
		['<r', '<r/>'],
	])('invalid xml node "%s"', (input, expected) => {
		const { errors, parser } = getTestParser();

		const actual = parser.parseFromString(input, 'text/xml').documentElement.toString();

		expect({ actual, ...errors }).toMatchSnapshot({ actual: expected });
	});

	it('html attribute (miss quote)', () => {
		const { errors, parser } = getTestParser();

		const actual = parser.parseFromString('<img attr=1/>', 'text/html').toString();

		expect({ actual, ...errors }).toMatchSnapshot();
	});

	it.each(['text/html', 'text/xml'])('%s attribute (missing =)', (mimeType) => {
		const { errors, parser } = getTestParser();
		const xml = [
			'<scxml xmlns="http://www.w3.org/2005/07/scxml" version="1.0"',
			'       profile="ecmascript" id="scxmlRoot" initial="start">',
			'',
			'  <!--',
			'      some comment (next line is empty)',
			'',
			'  -->',
			'',
			'  <state id="start" name="start">',
			// this line contains the missing = for attribute value
			'    <transition event"init" name="init" target="main_state" />',
			'  </state>',
			'',
			'  </scxml>',
		].join('\n');

		const actual = parser.parseFromString(xml, mimeType).toString();

		expect({ actual, ...errors }).toMatchSnapshot();
	});
});
