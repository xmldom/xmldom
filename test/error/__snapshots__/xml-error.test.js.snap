// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`html vs xml: html attribute (miss quote) 1`] = `
Object {
  "actual": "<img attr=\\"1\\" xmlns=\\"http://www.w3.org/1999/xhtml\\"/>",
  "warning": Array [
    "[xmldom warning]	attribute \\"1\\" missed quot(\\")!
@#[line:1,col:1]",
  ],
}
`;

exports[`html vs xml: invalid xml node "<r" 1`] = `
Object {
  "actual": "<r/>",
  "error": Array [
    "[xmldom error]	unexpected end of input
@#[line:1,col:1]",
  ],
  "warning": Array [
    "[xmldom warning]	unclosed xml attribute
@#[line:1,col:1]",
  ],
}
`;

exports[`html vs xml: invalid xml node "<test><!--" 1`] = `
Object {
  "actual": "<test/>",
  "error": Array [
    "[xmldom error]	Unclosed comment
@#[line:1,col:7]",
  ],
  "warning": Array [
    "[xmldom warning]	unclosed xml attribute
@#[line:1,col:1]",
  ],
}
`;

exports[`html vs xml: text/html attribute (missing =) 1`] = `
Object {
  "actual": "<scxml xmlns=\\"http://www.w3.org/2005/07/scxml\\" version=\\"1.0\\" profile=\\"ecmascript\\" id=\\"scxmlRoot\\" initial=\\"start\\">

  <!--
      some comment (next line is empty)

  -->

  <state id=\\"start\\" name=\\"start\\">
    <transition event=\\"init\\" name=\\"init\\" target=\\"main_state\\"/>
  </state>

  </scxml>",
  "warning": Array [
    "[xmldom warning]	attribute value must after \\"=\\"
@#[line:10,col:5]",
  ],
}
`;

exports[`html vs xml: text/xml attribute (missing =) 1`] = `
Object {
  "actual": "<scxml xmlns=\\"http://www.w3.org/2005/07/scxml\\" version=\\"1.0\\" profile=\\"ecmascript\\" id=\\"scxmlRoot\\" initial=\\"start\\">

  <!--
      some comment (next line is empty)

  -->

  <state id=\\"start\\" name=\\"start\\">
    <transition event=\\"init\\" name=\\"init\\" target=\\"main_state\\"/>
  </state>

  </scxml>",
  "warning": Array [
    "[xmldom warning]	attribute value must after \\"=\\"
@#[line:10,col:5]",
  ],
}
`;

exports[`html vs xml: unclosed document in text/html 1`] = `
Object {
  "actual": "<img xmlns=\\"http://www.w3.org/1999/xhtml\\"/>",
}
`;

exports[`html vs xml: unclosed document in text/xml 1`] = `
Object {
  "actual": "<img/>",
  "warning": Array [
    "[xmldom warning]	unclosed xml attribute
@#[line:1,col:1]",
  ],
}
`;
